### PR TITLE
kubeadm: remove the --experimental-kustomize feature

### PR DESCRIPTION
**Pull request by neolit123**
**What this PR does / why we need it**:

The feature was replaced by --experimental-patches and deprecated in
1.19.

Second commit updates go.mod and vendor/modules.txt.

**Which issue(s) this PR fixes**:

xref https://github.com/kubernetes/kubeadm/issues/2046
xref https://github.com/kubernetes/enhancements/pull/1991

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
action-required: kubeadm: remove the deprecated feature --experimental-kustomize from kubeadm commands. The feature was replaced with --experimental-patches in 1.19. To migrate see the --help description for the --experimental-patches flag.
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs
- [KEP]: https://github.com/kubernetes/enhancements/tree/master/keps/sig-cluster-lifecycle/kubeadm/1739-customization-with-patches
```

### CODE REVIEW

**Comment by neolit123** [comment]
the removal was announced on the SIG CL mailing list as well:
https://groups.google.com/g/kubernetes-sig-cluster-lifecycle/c/bv8ZbpmO5Q8

**Review by fabriziopandini — COMMENTED** [review]
@neolit123 thanks for this PR!
Changes seem pretty straight forward to me. Ready to approve as soon a WIP is removed

**Comment by fabriziopandini** [comment]
/lgtm
/hold
@yagonobre do you mind taking a look at this PR?

**Comment by yagonobre** [comment]
I'll take a look in a bit

File: vendor/modules.txt [hunk 1]
```diff
@@ -2493,7 +2493,6 @@ k8s.io/utils/trace
 sigs.k8s.io/apiserver-network-proxy/konnectivity-client/pkg/client
 sigs.k8s.io/apiserver-network-proxy/konnectivity-client/proto/client
 # sigs.k8s.io/kustomize v2.0.3+incompatible => sigs.k8s.io/kustomize v2.0.3+incompatible
-## explicit
```
Comment (yagonobre) [inline vendor/modules.txt:2496]: are we removing it intentionally?

**Review by yagonobre — COMMENTED** [review]
Feel free to remove the hold as soon as you want.
/lgtm

File: vendor/modules.txt (hunk 1 above)
Comment (neolit123) [reply vendor/modules.txt:2496]: this particular change was the result of ./hack/update-vendor.sh.
kustomize still remains vendored for usage in kubectl.

**Comment by neolit123** [comment]
/assign @liggitt 
PTAL for approval on the go.mod/vendor changes.

File: vendor/modules.txt (hunk 1 above)
Comment (liggitt) [reply vendor/modules.txt:2496]: this means k8s.io/kubernetes doesn't have a direct requirement for this module, but still has indirect requirements for it via various other modules. this is expected.

**Comment by liggitt** [comment]
/approve
for go.mod / vendor changes

/hold for approve from kubeadm reviewers

**Comment by liggitt** [comment]
oops, missed https://github.com/kubernetes/kubernetes/pull/94871#pullrequestreview-495580383 already lgtmed.

/hold cancel

**Comment by neolit123** [comment]
thanks

**Pull request merged**